src/storage.ts

```diff
@@ -550,6 +626,132 @@ export class Storage extends Service {
+   *     HMAC key is created for.
+   * @property {string} state The state of this HMAC key. One of "ACTIVE",
+   *     "INACTIVE" or "DELETED".
+   * @property {string} timeCreated The creation time of the HMAC key in
+   *     RFC 3339 format.
+   * @property {string} [updated] The time this HMAC key was last updated in
+   *     RFC 3339 format.
+   */
+  /**
+   * @typedef {object} HmacKeyResource
+   * @property {HmacKey} hmacKey The HmacKey object.
+   * @property {string} secret The HMAC key secret used to access XML API.
+   */
+  /**
+   * @typedef {array} CreateHmacKeyResponse
+   * @property {HmacKeyResource} 0 The HMAC key resource.
+   */
+  /**
+   * @callback CreateHmacKeyCallback Callback function.
+   * @param {?Error} err Request error, if any.
+   * @param {HmacKeyResource} resource An object containing the HMAC key's secret and the HmacKey object.
+   * @param {object} apiResponse The raw API response.
+   */
+  /**
+   * Create an HMAC key associated with an service account to authenticate
+   * requests to the Cloud Storage XML API.
+   *
+   * @see [HMAC keys documentation]{@linkhttps://cloud.google.com/storage/docs/authentication/hmackeys}
+   *
+   * @param {string} serviceAccountEmail The service account's email address
+   *     with which the HMAC key is created for.
+   * @param {CreateHmacKeyCallback} [callback] Callback function.
+   * @return {Promise<CreateHmacKeyResponse>}
+   *
+   * @example
+   * const {Storage} = require('google-cloud/storage');
+   * const storage = new Storage();
+   *
+   * // Replace with your service account's email address
+   * const serviceAccountEmail =
+   *   'my-service-account@appspot.gserviceaccount.com';
+   *
+   * storage.createHmacKey(serviceAccountEmail, function(err, hmacKeyResource) {
+   *   if (!err) {
+   *     const secret = hmacKey.secret;
+   *     // Securely store the secret for use with the XML API.
+   *   }
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * storage.createHmacKey(serviceAccountEmail)
+   *   .then((response) => {
+   *     const [hmacKeyResource] = response;
+   *     const secret = hmacKeyResource.secret;
+   *     // Securely store the secret for use with the XML API.
+   *   });
+   */
+  createHmacKey(
+    serviceAccountEmail: string,
+    optionsOrCb?: CreateHmacKeyOptions | CreateHmacKeyCallback,
+    cb?: CreateHmacKeyCallback
+  ): Promise<CreateHmacKeyResponse> | void {
+    if (typeof serviceAccountEmail !== 'string') {
+      throw new Error(
+        'The first argument must be a service account email to create an HMAC key.'
+      );
+    }
+
+    const {options, callback} = normalize<
+      CreateHmacKeyOptions,
+      CreateHmacKeyCallback
+    >(optionsOrCb, cb);
+    const query = Object.assign({}, options, {serviceAccountEmail});
+
+    this.request(
+      {
+        method: 'POST',
+        uri: `/projects/${this.projectId}/hmacKeys`,
```

Comment:
I think we need to get the project ID from the auth client first. Or does this still do the placeholder find and replace?

Reply:
Yes, I expect the placeholder replacement would work here. Any concerns that it might not?

Reply:
Actually, looking at this again, what is the full URL you are posting to?

Reply:
`https://www.googleapis.com/storage/v1/projects/{{projectId}}/hmacKeys`

Reply:
Gotcha, thanks. This should work as expected. Sorry for my confusion!